src/safemode_ui.cpp

```diff
@@ -0,0 +1,741 @@
+#include "safemode_ui.h"
+
+#include "game.h"
+#include "player.h"
+#include "output.h"
+#include "debug.h"
+#include "catacharset.h"
+#include "translations.h"
+#include "cata_utility.h"
+#include "path_info.h"
+#include "filesystem.h"
+#include "input.h"
+#include "mtype.h"
+#include "generic_factory.h"
+#include "worldfactory.h"
+#include "monstergenerator.h"
+#include "debug.h"
+
+#include <stdlib.h>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <locale>
+
+safemode &get_safemode()
+{
+    static safemode single_instance;
+    return single_instance;
+}
+
+void safemode::show()
+{
+    show( _( " SAFEMODE MANAGER " ), true );
+}
+
+void safemode::show( const std::string &custom_name, bool is_safemode )
+{
+    auto vRulesOld = vRules;
+
+    const int iHeaderHeight = 4;
+    const int iContentHeight = FULL_SCREEN_HEIGHT - 2 - iHeaderHeight;
+
+    const int iOffsetX = ( TERMX > FULL_SCREEN_WIDTH ) ? ( TERMX - FULL_SCREEN_WIDTH ) / 2 : 0;
+    const int iOffsetY = ( TERMY > FULL_SCREEN_HEIGHT ) ? ( TERMY - FULL_SCREEN_HEIGHT ) / 2 : 0;
```

Comment:
Are you copy+pasting the UI from somewhere?

It would be a good idea to clean it up. For example, make it cover the whole screen instead of just the middle.